src/plugins/install/index.rst

```diff
@@ -1,4 +1,162 @@
 Pre-Configure And Post-Install Plugins
 ======================================
 
-.. automodule:: cylc.flow.install_plugins
+Pre-configure plugins allow Cylc to take additional actions before running
+Cylc utilities such as ``cylc install``, ``cylc graph`` and ``cylc config``.
+
+Post-install plugins allow Cylc to take additional actions after
+running ``cylc install``.
+
+
+Built In Plugins
+----------------
+
+Cylc Flow provides the following pre-configure and post-install plugins:
+
+.. autosummary::
+   :toctree: built-in
+   :template: docstring_only.rst
+
+   cylc.flow.install_plugins.log_vc_info
+
+.. Note: Autosummary generates files in this directory, these are cleaned
+         up by `make clean`.
+
+Developing ``pre_configure`` and ``post_install`` plugins
+---------------------------------------------------------
+
+Cylc uses entry points registered by setuptools to search for pre-configure
+and post-install plugins.
+
+Hello World
+^^^^^^^^^^^
+
+In this example a pre-configure plugin which logs a "Hello World" message
+and, after installation, logs some info about the installation:
+
+.. code-block:: python
+   :caption: my_plugin.py
+
+   from cylc.flow import LOG
+
+   def pre_configure(srcdir=None, opts=None, rundir=None):
+      # write Hello to the Cylc log.
+      LOG.info(f'Hello World')
+      return {}
+
+   def post_install(srcdir=None, opts=None, rundir=None):
+      LOG.info(f'installed from {srcdir}')
+      LOG.info(f'installed to {rundir}')
+      LOG.info(f'installation options were {options}')
+      return {}
+
+Plugins are registered by registering them with the ``cylc.pre_configure``
+and ``cylc.post_install`` entry points:
+
+.. code-block:: python
+   :caption: setup.py
+
+   # plugins must be properly installed, in-place PYTHONPATH meddling will
+   # not work.
+
+   from setuptools import setup
+
+   setup(
+       name='my-plugin',
+       version='1.0',
+       py_modules=['my_plugin'],
+       entry_points={
+          # register this plugin with Cylc
+          'cylc.pre_configure': [
+            # name = python.namespace.of.module
+            'my_plugin=my_plugin.my_plugin:pre_configure'
+          ]
+          'cylc.post_install': [
+            # name = python.namespace.of.module
+            'my_plugin=my_plugin.my_plugin:post_install'
+          ]
+       }
+   )
```

Comment:
Python indentation not quite right for these examples

Reply:
Gah!